### PR TITLE
Make re-exec mode work with differently named components

### PR DESCRIPTION
**Pull request by aw-altiscale**
re-exec mode is hard-coded in lots of way.  It shouldn't be.

### CODE REVIEW

**Comment by aw-altiscale** [comment]
This isn't perfect, but...

File: precommit/main/test-patch.sh [hunk 1]
```diff
@@ -555,17 +558,18 @@ function echo_and_redirect
...
+## @param       path
+## @returns     1 - no, path
+## @returns     0 - yes, path - BASEDIR
+function relative_dir
 {
-  local p=${PATCH_DIR#${BASEDIR}}
+  local p=${1#${BASEDIR}}
 
-  if [[ ${#p} -eq ${#PATCH_DIR} ]]; then
+  if [[ ${#p} -eq ${#BINDIR} ]]; then
```
Comment (sekikn) [inline precommit/main/test-patch.sh]: Should it be ```-eq ${#1}```?

Comment (aw-altiscale) [inline precommit/main/test-patch.sh]: Actually, this should probably be BASEDIR.  But I have a suspicion that it needs to take a second parameter which should get placed here.  I'll trace through the code when I get a chance.

Comment (aw-altiscale) [inline precommit/main/test-patch.sh]: Ha, no, you are correct. 1 will work here.

**Pull request merged**